File: Algorithm/Programmers/스택/크레인인형뽑기게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        int len = board.length;
        int[][] _board = new int[len][len];
        
        for(int i = 0; i<len; i++){
            for(int j = 0; j<len; j++){
                _board[i][j] = board[j][i];
            }
        }

        
        Stack<Integer> basket = new Stack<Integer>();
        for(int m : moves){
            for(int i = 0; i<len; i++){
                if(_board[m-1][i] != 0){
                    if(!basket.isEmpty() && (basket.peek()==_board[m-1][i])){
                        basket.pop();
                        answer+=2;
                    }else{
                        basket.push(_board[m-1][i]);                        
                    }                
                    _board[m-1][i] = 0;
                    break;
                }
            }
        }
        
        
        return answer;
    }
}

File: Algorithm/Programmers/Lv1/크레인 인형뽑기 게임.js
function solution(board, moves) {
    var answer = 0;
    let cnt = 0;
    let basket = [];
    board = board[0].map((_, colIndex) => board.map(row => row[colIndex]));
    let b = board.filter((x)=>{
       for(let i=0; i<board.length; i++){
           if(x[0]===0){
               x.shift();
           }
       }
        return x;
    });

    let doll;
    moves.forEach(function(m){
        if(b[m-1].length!==0){
            doll = b[m-1].shift();
            if(doll === basket[basket.length-1]){
                cnt+=2;
                basket.pop();
            }else{
                basket.push(doll);    
            }   
        }
    });
    return cnt;
}
